packages/react-native/ReactAndroid/src/test/java/com/facebook/react/modules/timing/TimingModuleTest.kt

```diff
@@ -0,0 +1,257 @@
+    verifyNoMoreInteractions(jSTimersMock)
+  }
+
+  @Test
+  fun testPausingAndResuming() {
+    timingModule.onHostResume()
+    timingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHostPause()
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(jSTimersMock)
+    reset(jSTimersMock)
+    timingModule.onHostResume()
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+  }
+
+  @Test
+  fun testHeadlessJsTaskInBackground() {
+    timingModule.onHostPause()
+    timingModule.onHeadlessJsTaskStart(42)
+    timingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(jSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskInForeground() {
+    timingModule.onHostResume()
+    timingModule.onHeadlessJsTaskStart(42)
+    timingModule.createTimer(41.0, 1.0, 0.0, true)
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHostPause()
+    verifyNoMoreInteractions(jSTimersMock)
+  }
+
+  @Test
+  fun testHeadlessJsTaskIntertwine() {
+    timingModule.onHostResume()
+    timingModule.onHeadlessJsTaskStart(42)
+    timingModule.createTimer(41.0, 1.0, 0.0, true)
+    timingModule.onHostPause()
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHostResume()
+    timingModule.onHeadlessJsTaskFinish(42)
+    stepChoreographerFrame()
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(41.0))
+    reset(jSTimersMock)
+    timingModule.onHostPause()
+    stepChoreographerFrame()
+    verifyNoMoreInteractions(jSTimersMock)
+  }
+
+  @Test
+  fun testSetTimeoutZero() {
+    timingModule.createTimer(100.0, 0.0, 0.0, false)
+    verify(jSTimersMock).callTimers(JavaOnlyArray.of(100.0))
+  }
+
+  @Test
+  fun testActiveTimersInRange() {
+    timingModule.onHostResume()
+    assertThat(timingModule.hasActiveTimersInRange(100)).isFalse
+    timingModule.createTimer(41.0, 1.0, 0.0, true)
+    assertThat(timingModule.hasActiveTimersInRange(100)).isFalse // Repeating
+    timingModule.createTimer(42.0, 150.0, 0.0, false)
+    assertThat(timingModule.hasActiveTimersInRange(100)).isFalse // Out of range
+    assertThat(timingModule.hasActiveTimersInRange(200)).isTrue // In range
+  }
+
+  @Test
+  @Ignore("Failing due to failed invocation verification") // TODO T13905097
+  fun testIdleCallback() {
+    timingModule.onHostResume()
+    timingModule.setSendIdleEvents(true)
```

Comment:
@cortinico Just realised this test will pass if we swap these two statements. Should we include it in this PR?